### DISCUSSION TITLE
Why are there performance differences between dongles running same firmware?

### DISCUSSION THREAD

**Discussion by dmavromatis** [ludwig-v/wireless-carplay-dongle-reverse-engineering#31] [Q&A]
I had asked a question under issues — but this is probably a more appropriate space to ask.  

I have both a Carlinkit Plus and a C2Play dongle — the C2Play one seems to be more responsive.  Any thought on why this would be the case running the exact same firmware?  Could it be QA/QC issues?

**Comment by ghost** [comment]
I have a Carlinkit 2 and I don’t see any issues regarding responsiveness. 

 Have my FPS set to 60 (not sure if this actually does anything. 
 
I would even go as far to saw my experience is fast and snappy. 
 
Running Feb 2021 Custom

**Reply by mautorresp** [reply]
Hi, I seem to be unable to put any firmware in it, it just stays solid red, how did you manage to put the firmware?

**Comment by milesking** [comment]
The variant I got is CPC200-U2W, the searching UI is laggy especially the language button which takes over 3s to respond. But the CarPlay UI works well with expected wireless latency. I noticed after connect/disconnect iPhone the HU's CarPlay UI kind of switching its working status.

**Comment by dmavromatis** [comment]
I ordered another CPC200-U2W Plus and that performs better.  The other must have some hardware/antenna issue.  Thanks for the confirmation.
